scapy/contrib/cdp.py

```diff
@@ -357,7 +357,7 @@ def _check_len(self, pkt):
         This padding is only used for checksum computation.  The original
         packet should not be altered."""
         if len(pkt) % 2:
-            last_chr = pkt[-1]
+            last_chr = pkt[len(pkt) - 1:]
```

Comment:
How is this related to the change of the field?

Reply:
This is a separate change.  I need to work through the details of the pull request workflow.  I will fix.